h2/src/main/org/h2/mvstore/db/MVTable.java

```diff
@@ -564,6 +564,7 @@ public void updateRow(Session session, Row oldRow, Row newRow) {
 
     @Override
     public Row lockRow(Session session, Row row) {
+        syncLastModificationIdWithDatabase();
```

Comment:
Taking a lock is not a data modification, so why do we need it here? Taking a lock should not prevent anybody from using cached result of a previous query etc.

Reply:
Lock means that we can update our snapshot with actual data from another committed transaction, without this some new tests from this PR return invalid results. See also `SELECT FOR UPDATE` documentation.

Reply:
I guess it could be optimized and invoked only when row was modified by the lock.

Reply:
In that way such check shouldn't be very expensive:
https://github.com/h2database/h2database/blob/4168dddc1da2943d0d1ac10208004441dd2edb2e/h2/src/main/org/h2/command/dml/Select.java#L435